tardis/utilities/utils.py

```diff
@@ -66,6 +67,32 @@ def csv_parser(
             }
 
 
+KeyType = TypeVar("KeyType")
+
+
+def deep_update(
+    original_mapping: Mapping[KeyType, Any],
+    mapping_update: Mapping[KeyType, Any],
+) -> Mapping[KeyType, Any]:
+    from collections.abc import Mapping
```

Comment:
The types in `typing` are thin wrappers around the ABCs in `collection.abc`; you don't need to import the `Mapping` ABC since you can just use the type.

For example:
```python3
>>> import typing
>>> isinstance({}, typing.Mapping)
True
```

```suggestion
```

Reply:
Thanks, did not know that it is just wrapper.